src/options.rs

```diff
@@ -136,25 +140,50 @@ impl Options {
     )
   }
 
-  pub(crate) fn bitcoin_rpc_client(&self) -> Result<Client> {
-    let cookie_file = self
-      .cookie_file()
-      .map_err(|err| anyhow!("failed to get cookie file path: {err}"))?;
+  fn derive_var(
+    arg: Option<OsString>,
+    env: Option<OsString>,
+    config: Option<OsString>,
+    default: Option<OsString>,
+  ) -> Option<OsString> {
+    arg.or(env).or(config).or(default)
+  }
 
-    let rpc_url = self.rpc_url();
+  pub(crate) fn auth(&self) -> Auth {
+    log::info!("Connecting to Bitcoin Core at {}", self.rpc_url());
+
+    let config = self.load_config().unwrap();
```

Comment:
This function must return an error, and we should include here a good error message. There are many reasons why config loading might fail, so this will be a common error that users encounter.